CMakeLists.txt

```diff
@@ -63,26 +54,10 @@ endif()
 # ----------------------------------------
 # use C++14 and C99
 # see http://stackoverflow.com/questions/10851247/how-to-activate-c-11-in-cmake
-include(CheckCXXCompilerFlag)
-include(CheckCCompilerFlag)
-CHECK_CXX_COMPILER_FLAG("-std=c++14" COMPILER_SUPPORTS_CXX14)
-CHECK_CXX_COMPILER_FLAG("-std=c++0x" COMPILER_SUPPORTS_CXX0X)
-CHECK_CXX_COMPILER_FLAG("-fPIC" COMPILER_SUPPORTS_FPIC)
-if (COMPILER_SUPPORTS_CXX14)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
-elseif(COMPILER_SUPPORTS_CXX0X)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
-else()
-    message( WARNING "The compiler ${CMAKE_CXX_COMPILER} doesn't support the -std=c++14 flag. Some code may not compile.")
-endif()
-
-CHECK_C_COMPILER_FLAG("-std=c99" COMPILER_SUPPORTS_C99)
-if (COMPILER_SUPPORTS_C99)
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
-else()
-    message( WARNING "The compiler ${CMAKE_C_COMPILER} doesn't support the -std=c99 flag. Some code may not compile.")
-endif()
-
```

Comment:
This change makes sense. I would prefer spaces inside functions. Again, our CMake isn't entirely consistent, but most calls have spaces to improve readability:
```
set( CMAKE_C_STANDARD 99 )
```
However, I think it makes sense to require C99 and C++14. Are there any current compilers that don't support those standards?

Reply:
Regarding my last comment, I changed the CMakeLists.txt to require c99 and c++14. I really don't understand CMake's motivation for having (effectively) `CMAKE_CXX_STANDARD_REQUIRED = off` and `CMAKE_CXX_EXTENSIONS = on` by default.